### PR TITLE
fix(core): Add lint check for core

### PR DESCRIPTION
**Pull request by petamoriken**
The prefer-primordials lint was skipped for `core/*.js`.

### CODE REVIEW

File: tools/lint.js [hunk 1]
```diff
@@ -66,7 +66,8 @@ async function dlintPreferPrimordials() {
   const sourceFiles = await getSources(ROOT_PATH, [
     "runtime/**/*.js",
     "ext/**/*.js",
-    "core/**/*.js",
+    "core/*.js",
```
Comment (petamoriken) [inline tools/lint.js:69]: I don't know why `core/*.js` is OK but `core/**/*.js` is NG.

Comment (petamoriken) [reply tools/lint.js:69]: Apparently globstar (`**`) has not agreed on the behavior and `git` requires one directory.

File: core/01_core.js [hunk 2]
```diff
@@ -235,7 +242,7 @@
   }
 
   function opAsync(opName, ...args) {
-    return ops[opName](...args);
+    return ops[opName](...new SafeArrayIterator(args));
```
Comment (ry) [inline core/01_core.js:245]: Thanks for the patch but this likely has negative performance implications.

can you check the before and after of op_boop bench?

Comment (petamoriken) [reply core/01_core.js:249]: Using `ReflectApply` seemed to perform better than using `SafeArrayIterator`.

main branch

```
running 3 tests
test bench_op_async   ... bench:     349,345 ns/iter (+/- 7,851)
test bench_op_nop     ... bench:      37,275 ns/iter (+/- 2,333)
test bench_op_pi_json ... bench:      41,590 ns/iter (+/- 2,003)
```

use `SafeArrayIterator`

```
running 3 tests
test bench_op_async   ... bench:     460,511 ns/iter (+/- 8,330)
test bench_op_nop     ... bench:      36,833 ns/iter (+/- 2,021)
test bench_op_pi_json ... bench:      41,529 ns/iter (+/- 2,572)
```

use `ReflectApply`

```
running 3 tests
test bench_op_async   ... bench:     348,736 ns/iter (+/- 11,100)
test bench_op_nop     ... bench:      37,114 ns/iter (+/- 2,677)
test bench_op_pi_json ... bench:      41,799 ns/iter (+/- 2,537)
```

**Review by littledivy — COMMENTED** [review]
We will be removing `opAsync` from core soon enough (The new way to async ops is simply `await ops.op_async()`) so maybe lets leave doing iterator primordial for it?

**Comment by petamoriken** [comment]
@littledivy Could you merge this PR? Or do I need to wait for `opAsync` to be removed?

**Review by kt3k — APPROVED** [review]
LGTM

**Review by littledivy — APPROVED** [review]
LGTM!

**Pull request merged**